### DISCUSSION TITLE
Failed to load and keep a single Workout out of my "Workouts" array

### DISCUSSION THREAD

**Discussion by thomasweigert91** [neuefische/questions#400] [Web] [answered]
i have the problem that i can't access individual workouts through my dynamic routes. in the meantime it works, but only if i change the code (especially my const workout from workoutid === id to workoutid=id). As soon as I select a new workout, it throws me again only an empty object.

```
const Workout = () => {
  const router = useRouter();
  const {id} = router.query;
  console.log(id);

  const [currentWorkout, setCurrentWorkout] = useState(
    JSON.parse(localStorage.getItem("workouts") || "[]")
  );

  useEffect(() => {
    const getWorkoutData = async () => {
      const workouts = JSON.parse(localStorage.getItem("workouts"));

      const workout = workouts.find(workout => workout.id === id);
      console.log(currentWorkout);
      setCurrentWorkout(workout);
    };
    getWorkoutData();
  }, [id]);

```

**Comment by mihaimazareanu** [comment]
Hi Thomas,

I think your currentWorkout state variable should look like this:

    const [currentWorkout, setCurrentWorkout] = useState(
    JSON.parse(localStorage.getItem("workouts") || [])
    );
    
The way you have written your code, if there is nothing in `localStorage` your initial value would be the string `"[]"`

However, I don't understand why would you want `currentWorkout` 's initial state to be `workouts` from `localStorage`? `currentWorkout` seems to me to be an object, while `workouts` is an array of objects, if I understand your code correctly.

So I would set `currentWorkout` to `null` initially and then use `useEffect` to set currentWorkout to that which you have filtered out with your `id`.

I hope this helps...

**Comment by thomasweigert91** [comment, marked as answer]
The easiest solution was to delete the useEffect and just pass the data from the localstorage into a simple state.
